### ISSUE TITLE
Prop changes doesn't proc rerender inside ReactCSSTransitionReplace children

### ISSUE THREAD

**Issue by are1000** [marnusw/react-css-transition-replace#53]
I'm using ReactCSSTransitionReplace in my application. Function `this.getMenu` returns a different menu depending on the `currentUnit` and `data` and when the menu changes it is animated (this is expected).
    
    return <ReactCSSTransitionReplace
      style={{ height: '100%' }}
      transitionName={this.isSelected() ? slideLeft : slideRight}
      transitionEnterTimeout={400}
      transitionLeaveTimeout={400}
    >
      {this.getMenu(this.props.currentUnit, this.props.data)}
    </ReactCSSTransitionReplace>

But when I update the `currentUnit` prop on my component, menus shouldn't change - inside the current menu the highlighted entry should change, but it should not trigger an animation (which is also expected). But the problem is - while the prop successfully propagates inside the `getMenu` function (when I put a `console.log` there it receives new prop), the changes doesn't reflect in the browser - nothing changes. It has to be an issue with RCTR because when I remove it, the problem doesn't occur. Maybe it implements sCU somewhere and it doesn't catch the changing prop? Is there a workaround?

Edit 1: The issue is still there in `3.0.1`.

**Comment by GeKorm** [comment]
I've found that it only updates for different keys. So for now you can either revert to <3.0.0 which I've found to not be compatible with React 16, or if you only have a couple props you can calculate the key from them. For example 
```javascript
<div key=`keyname${someArray.length}${aBoolean}${Object.keys(anotherProp)}` />
```

Just a temporary solution until a fix is published.

**Comment by are1000** [comment]
@GeKorm The problem is I don't want to trigger the animation, just change some `classNames` way inside the RCTR children.

**Comment by marnusw** [comment]
@GeKorm are you saying this used to work fine in v2, but it's no longer working in v3?

**Comment by are1000** [comment]
@marnusw I can confirm that it is indeed working in `2.2.1`, but not in `3.0.1`.

**Comment by GeKorm** [comment]
I've created v2.2.2 until v3 is more stable (#55 if you want to publish it @marnusw)
It contains compatibility changes for React 16.

You can use it by doing:
`yarn add https://github.com/GeKorm/react-css-transition-replace`  
or
`npm i https://github.com/GeKorm/react-css-transition-replace`
will replace it and let you use the fork.

**Comment by marnusw** [comment]
v2.2.2 has been published, but I also plan to fix this over the weekend. I'm pretty sure I know why this is happening.

**Comment by marnusw** [comment]
I'm pretty sure this will be fixed in `v3.0.2` just released, but let me know if there are any further issues.

**Comment by awreccan** [comment]
Rerendering works in `v3.0.2`, thanks!
